EIPS/eip-6982.md

```diff
@@ -0,0 +1,80 @@
+---
+eip: 6982
+title: Efficient Default Lockable Tokens
+description: A gas-efficient approach to lockable ERC-721 tokens
+author: Francesco Sullo (@sullof), Alexe Spataru (@urataps)
+discussions-to: https://ethereum-magicians.org/t/erc721-default-lockable-proposal/13366
+status: Review
+type: Standards Track
+category: ERC
+created: 2023-05-02
+requires: 165, 721
+---
+
+## Abstract
+
+This proposal introduces a lockable interface for [ERC-721](./eip-721.md) tokens that optimizes gas usage by eliminating unnecessary events. This interface forms the foundation for the creation and management of lockable [ERC-721](./eip-721.md) tokens. It provides a gas-efficient approach by emitting a `DefaultLocked(bool locked)` event upon deployment, setting the initial lock status for all tokens, while individual `Locked(uint256 indexed tokenId, bool locked)` events handle subsequent status changes for specific tokens. The interface also includes a view function `locked(uint256 tokenId)` to return the current lock status of a token.
+
+## Motivation
+
+Existing lockable token proposals often mandate the emission of an event each time a token is minted. This results in unnecessary gas consumption, especially in cases where tokens are permanently locked from inception to destruction (e.g., soulbounds or non-transferable badges). This proposal offers a more gas-efficient solution that only emits events upon contract deployment and status changes of individual tokens.
+
+## Specification
+
+The interface is defined as follows:
+
+```solidity
+// erc165 interfaceId 0x6b61a747
+interface IERC6982 {
+  // MUST be emitted one time, when the contract is deployed,
+  // defining the default status of any token that will be minted
+  event DefaultLocked(bool locked);
+
+  // MUST be emitted any time the status changes
+  event Locked(uint256 indexed tokenId, bool locked);
+
+  // Returns the default status of the token.
+  function defaultLocked() external view returns (bool);
+
+  // Returns the status of the token.
+  // If no special event occurred, it MUST return what defaultLocked() returns.
+  // It MUST revert if the token does not exist.
+  function locked(uint256 tokenId) external view returns (bool);
+}
+```
+
+The [ERC-165](./eip-165.md) interfaceId is `0xb45a3c0e`.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
```

Comment:
Please move this to the top of the specification section.

Reply:
Done.